processor/deltatocumulativeprocessor/internal/data/expo/scale.go

```diff
@@ -103,13 +103,5 @@ func Collapse(bs Buckets) {
 			counts.SetAt(i, counts.At(k)+counts.At(k+1))
 		}
 	}
-
-	// zero the excess area. its not needed to represent the observation
-	// anymore, but kept for two reasons:
-	// 1. future observations may need it, no need to re-alloc then if kept
-	// 2. [pcommon.Uint64Slice] can not, in fact, be sliced, so getting rid
-	//    of it would alloc ¯\_(ツ)_/¯
-	for i := size; i < counts.Len(); i++ {
-		counts.SetAt(i, 0)
-	}
+	counts.FromRaw(counts.AsRaw()[:size])
```

Comment:
This is a tricky one.

This change makes us reallocate all bucket counts every time we downscale, regardless of whether we are close to the limit or not.
Worse, it allocs twice because `AsRaw` copies, and `FromRaw` copies again.

Generally, we should keep allocated memory even if we have no counts in there, because those might arrive in the future. This is fine because we enfore our limit by downscaling before, so we will never exceed it.

Reply:
IIRC I had to do this because we don't explicitly store the upper bound of the bucket, so we rely on the [length of the slice itself](https://github.com/edma2/opentelemetry-collector-contrib/blob/022c83f96966a4ab3a9e5fd5acf472d3f3821ecf/processor/deltatocumulativeprocessor/internal/data/expo/expo.go#L41). We must know the upper bound of the bucket to calculate the new scale. We could technically store this information somewhere but it added more complexity.

FWIW I didn't see any noticeable hit to performance when this optimization was removed, but I am happy to follow a different suggestion.